### PR TITLE
PLAT-707 Add kafka integration

### PR DESCRIPTION
**Pull request by nour-borgi**

### CODE REVIEW

File: app/scripts/controllers/channel-tabs/routes.js [hunk 1]
```diff
@@ -277,6 +302,37 @@ export function channelRoutesCtrl ($scope, $timeout, Api, Alerting) {
...
 
+  // check if topic name is valid by kafka
+  $scope.checkIskafkaTopicValid = function (value) {
+    if (value) {
+      if(value.length > 255) {
+        return 'Max length is 255 characters!'
+      }
+      for(let char of value) {
+        if(!/[a-zA-Z0-9\\._\\-]/.test(char)) {
+          return 'Not valid topic name! Only letters, numbers, . (dot), _ (underscore), - (minus) can use used!'
```
Comment (michaelloosen) [inline app/scripts/controllers/channel-tabs/routes.js:313]:
```suggestion
          return 'Not valid topic name! Only letters, numbers, . (dot), _ (underscore), - (minus) can be used!'
```

File: app/views/partials/channels-tab-routes.html [hunk 2]
```diff
@@ -55,49 +54,80 @@ <h3>Set Route</h3>
...
+        <label class="btn btn-primary" ng-model="newRoute.waitPrimaryResponse" uib-btn-radio="false">No</label>
+      </div>
     </div>
-    <div class="form-group" style="float: left; margin-left: 20px;">
+    <div class="form-group" style="float: left; margin-left: 20px;" ng-show="newRoute.waitPrimaryResponse==true" ng-class="{ 'has-error' : ngErrorRoute.statusCodesCheck }">
       <label>
-        <span style="color: #a94442;">**</span> Route Path Transform
-        <i ng-show="uiSettings.showTooltips" uib-tooltip-html="'Transform the route path by using the following substitute expression format:<br /> s/from/to/g'" class="glyphicon glyphicon-question-sign tooltip-wide"></i>
+        Send request when matching status code
+        <i ng-show="uiSettings.showTooltips" uib-tooltip="Request will be sent to the second route unless the response status of the primary route match one of these status codes. A single value or a list are both accepted. A list should be comma separated (single status codes e.g. 200,201, range e.g. 2**,3**,  mixture of both e.g. 204,3**)." class="glyphicon glyphicon-question-sign tooltip-wide" style="font-size: 12px"></i>
```
Comment (rcrichton) [inline app/views/partials/channels-tab-routes.html:68]:
```suggestion
        <i ng-show="uiSettings.showTooltips" uib-tooltip="Request will NOT be sent to the second route unless the response status of the primary route match one of these status codes. A single value or a list are both accepted. A list should be comma separated (single status codes e.g. 200,201, range e.g. 2**,3**,  mixture of both e.g. 204,3**)." class="glyphicon glyphicon-question-sign tooltip-wide" style="font-size: 12px"></i>
```

Comment (rcrichton) [inline app/views/partials/channels-tab-routes.html:68]:
```suggestion
        <i ng-show="uiSettings.showTooltips" uib-tooltip="Request will NOT be sent to the secondary route(s) unless the response status of the primary route match one of these status codes. A single value or a list are both accepted. A list should be comma separated (single status codes e.g. 200,201, range e.g. 2**,3**,  mixture of both e.g. 204,3**)." class="glyphicon glyphicon-question-sign tooltip-wide" style="font-size: 12px"></i>
```

**Comment by rcrichton** [comment]
Overall this appears to be working very well! Nice job all!

**Review by rcrichton — CHANGES_REQUESTED** [review]
Looks good make some comments above.

**Review by bradsawadye — APPROVED** [review]

**Review by bradsawadye — APPROVED** [review]

**Pull request merged**